control-plane/agents/src/bin/core/volume/operations_helper.rs

```diff
@@ -184,7 +191,10 @@ impl OperationGuardArc<VolumeSpec> {
         let result = self.attach_to_target(registry, replica).await;
         self.complete_update(registry, result, spec_clone).await?;
 
-        registry.volume(&state.uuid).await
+        Ok(SetReplicaResult::new(
+            registry.volume(&state.uuid).await?,
+            vg_guard,
```

Comment:
Why do we need to return this after we've already increased/decreased it?

Reply:
Right. The operation is completed, we should not.